### ISSUE TITLE
director/filter/hostgroups ignores hostgroups assigned via apply-rules on host variables

### ISSUE THREAD

**Issue by melaniebauer** [Icinga/icingaweb2#3673]
The filter director/filter/hostgroups in roles.ini does only show hosts that have the given hostgroup directly assigned (using director). Hosts that are assigned to a hostgroup because a host variable is set, are not shown. 

Example config, host and group names changed:

**/var/lib/icinga2/...../zones.d/director/hostgroup.conf:**
<line>object HostGroup "Testgroup" { 
<line> assign where host.vars.typ == "test
<line>}

vs.

**/var/lib/icinga2/..../zones.d/master/hosts.conf**
<line>object Host "myhost" {
<line>...
<line>groups = [ "Testgroup"]
<line>...
<line>}

**/etc/icingaweb2/roles.ini:**
<line>[ConfigTest]
<line>groups = "ConfigTest"
<line>permissions = "module/director, director/*, module_monitoring"
<line>monitoring/filter/objects = "_host_typ=test"
<line>director/filter/hostgroups = "Testgroup"

Being a member of the group "ConfigTest", I do not see a host in director where host.vars.typ is set to "test", only a host where I set the hostgroup in director explicitely to the group "Testgroup" (in the above example this would be "myhost"). In monitoring, all hosts of Testgroup show up.

Furthermore, I am getting an error when I click on the hostgroup "Testgroup" in director (to configure the details). Error message is: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'o.object_name' in 'where clause', query was: SELECT h.id FROM icinga_hostgroup AS h WHERE (id='68') AND (o.object_name IN....
As I have found in another issue, this is an error for a condition that should not be reached so I assume this is a subsequent error, I only include this for completeness sake.

## Your Environment

* Icinga Web 2 version and modules (System - About): 1.6.0
* Version used (`icinga2 --version`): r2.10.2-1
* Operating System and version: openSUSE Leap 15.0
* Enabled features (`icinga2 feature list`): api checker command compatlog ido-mysql livestatus mainlog notification statusdata syslog
* Config validation (`icinga2 daemon -C`): no errors
* One zone ("master") with two endpoints (redundant icinga2 server, director is only installed on one of them, they share one ido-DB)

**Comment by Thomas-Gelf** [comment]
Hi @melaniebauer,

Hostgroups assigned via Apply Rules should work fine, that's the main use-case this feature has been built for. Did you upgrade recently, have you been running an older version before this one, does slightly modifying and storing your Apply Rule once again change anything?

Of course also the Error should not occur, could you please share an (eventually anonymized) screenshot showing where you're clicking to reproduce this? Your `roles.ini` shows that you're allowing `director/*` while restricting access to a specific Hostgroup. Depending on where you click, that might explain the shown error. Restricting access to specific hosts has only been tested with a very limited set of permissions, like `director/hosts`.

It's impossible to seriously restrict someone equipped with `director/*`, as he could always somehow cheat on you. Could you please describe the use-case you had in mind when defining this role?

Thanks,
Thomas

**Comment by melaniebauer** [comment]
Oh that is bad for me, nice for you. I have been upgrading our icinga2 installation for years now, always to the latest version. But I have never before used filters on the access to director.

So I have newly configured this and as it did not work, I made the test configuration as easy as possible. That's why I have allowed * on director. It's the same if I allow only director/hosts (only that I do not see any more menu points in director). 

I do not have to send screenshots, it's more than easy: director-hosts shows only one host (the one I assigned the group to). 

Actually I want different admins to configure only their own hosts. In a non-hostile environment, it is not necessary to close all holes. Besides the final config will be more complicated.

Anyway, I will try to close the issue and continue to search a solution for myself. Thanks for your attention.

**Comment by Thomas-Gelf** [comment]
Wait, your main problem (hosts matching your rule not being shown) isn't solved, is it?

**Comment by melaniebauer** [comment]
No, it is not. But I guess I should ask in the monitoring forum, not open an issue here. Unfortunately I do not know what to ask as I just tried this configuration on my old test machine (a VM) with OpenSUSE Leap 42.3, Kernel 4.4.165-82-default, icinga2 version r2.9.2-1 and director 1.4.3. There it works fine.

**Comment by Thomas-Gelf** [comment]
Don't worry. Sure, it works fine for me, but this doesn't mean that there might be no bug. It doesn't work for you out of the box, so there must be an issue. I'd love to fix this, but I need your help to track it down.

**Comment by melaniebauer** [comment]
Thanks Thomas. 

Well... I upgraded my test machine step by step, first icinga*, than the OS, in order to find out when the problem arises. Now I have the same version on the test and the productive machine. And I was not able to reproduce the error on the test machine. 

The only difference was the groupname (as I did not have the new group on my test machine). So I renamed the group Z_DN into ConfzoneDN. And now it worked.

So the problem occurs here only with the hostname "Z_DN" - or "ZoneDN". Do you have problems with "Z"?  

For me it is solved, I have a workaround, you may close it. Maybe you would like to check on groupnames?
